modules/javafx.controls/src/main/java/javafx/scene/control/ContextMenu.java

```diff
@@ -226,46 +221,45 @@ public String getName() {
 
     /**
      * Shows the {@code ContextMenu} relative to the given anchor node, on the side
-     * specified by the {@code hpos} and {@code vpos} parameters, and offset
+     * specified by the {@code side} parameter, and offset
      * by the given {@code dx} and {@code dy} values for the x-axis and y-axis, respectively.
      * If there is not enough room, the menu is moved to the opposite side and
      * the offset is not applied.
      * <p>
-     * To clarify the purpose of the {@code hpos} and {@code vpos} parameters,
-     * consider that they are relative to the anchor node. As such, a {@code hpos}
-     * and {@code vpos} of {@code CENTER} would mean that the ContextMenu appears
-     * on top of the anchor, with the (0,0) position of the {@code ContextMenu}
-     * positioned at (0,0) of the anchor. A {@code hpos} of right would then shift
-     * the {@code ContextMenu} such that its top-left (0,0) position would be attached
-     * to the top-right position of the anchor.
-     * <p>
```

Comment:
I think it is worth adding a short replacement for this, explaining how the side parameter and delta values affect the location of the popup.

Reply:
While trying to come up with a good documentation I've detected a real change in behaviour in connection with the NodeOrientation of the anchor node.
Although this has never been documented, when NodeOrientation.RIGHT_TO_LEFT is set, the context menu was right-aligned for Side=TOP and Side=BOTTOM. 
Since I assume we don't want to change this behaviour I would now document it and adapt my patch accordingly. I've already written a test case like this:
```
    @Test public void test_position_withOrientation() throws InterruptedException {
        ContextMenu cm = createContextMenu(false);
        anchorBtn.setNodeOrientation(NodeOrientation.RIGHT_TO_LEFT);
        cm.show(anchorBtn, Side.TOP, 0, 0);

        Bounds anchorBounds = anchorBtn.localToScreen(anchorBtn.getLayoutBounds());
        Node cmNode = cm.getScene().getRoot();
        Bounds cmBounds = cm.getScene().getRoot().localToScreen(cmNode.getLayoutBounds());

        assertEquals(anchorBounds.getMaxX(), cmBounds.getMaxX(), 0.0);
        assertEquals(anchorBounds.getMinY(), cmBounds.getMaxY(), 0.0);
        anchorBtn.setNodeOrientation(NodeOrientation.LEFT_TO_RIGHT);
    }
```
which passes with the old implementation but (currently) fails with the new implementation.

Reply:
Oh wow. Further experimentation has shown, that if NodeOrientation.RIGHT_TO_LEFT is used on the anchor, then Side.LEFT used to make the menu appear on the **right** hand side and Side.RIGHT used to make the menu appear on the **left** hand side of the button, which really suprises me, since I didn't expect the side parameter to be relative.

Reply:
Ok, I just pushed all the changes necessary to keep the old behaviour for right-to-left node orientation.
If you would rather have the behaviour changed (it doesn't seem very intuitive to me...) just tell me and I can change the implementation / tests.

Reply:
Good catch. The current behavior is consistent with how RTL works for most things, so we do want to preserve that behavior.